src/software/firmware/srcs/screen.cpp

```diff
@@ -91,18 +97,69 @@ void displaySettings(uint16_t peakPressureMax,
     screen.print(message);
 }
 
+static uint8_t prevNbAlarmToPrint = 255;
+static uint8_t prevAlarmCodes[MAX_ALARMS_DISPLAYED] = {0};
+
+static bool hasAlarmInformationChanged(uint8_t p_alarmCodes[], uint8_t p_nbTriggeredAlarms) {
+    uint8_t nbAlarmToPrint = min(static_cast<uint8_t>(MAX_ALARMS_DISPLAYED), p_nbTriggeredAlarms);
+
+    bool hasChanged = false;
+    if (nbAlarmToPrint != prevNbAlarmToPrint) {
+        hasChanged = true;
+    } else {
+        for (uint8_t i = 0; i < nbAlarmToPrint; ++i) {
+            if (p_alarmCodes[i] != prevAlarmCodes[i]) {
+                hasChanged = true;
+                break;
+            }
+        }
+    }
+
+    if (hasChanged) {
+        prevNbAlarmToPrint = nbAlarmToPrint;
+        for (uint8_t i = 0; i < nbAlarmToPrint; ++i) {
+            prevAlarmCodes[i] = p_alarmCodes[i];
+        }
+    }
+
+    return hasChanged;
+}
+
 void displayAlarmInformation(uint8_t p_alarmCodes[], uint8_t p_nbTriggeredAlarms) {
-    screen.setCursor(0, 2);
-    screen.print("                    ");
-    screen.setCursor(0, 2);
+    if (!hasAlarmInformationChanged(p_alarmCodes, p_nbTriggeredAlarms)) {
```

Comment:
There is a risk of data not being displayed as expected if the line is overwritten somewhere else in the code.

Reply:
Indeed. Maybe put this sentence in a comment and let's accept the risks? 😛

Reply:
done